### ISSUE TITLE
Snapshots disappear from grub every time there is a big update

### ISSUE THREAD

**Issue by KarkanAlzwayed** [Antynea/grub-btrfs#356]
Every time I update my system and reboot, the snapshots disappear from grub. I have found out that I would need to update grub by running these commands for them to show up. 
```
sudo grub-mkconfig -o /boot/grub/grub.cfg
sudo grub-mkconfig
```
I have tried reinstalling grub-btrfs but that didn't help.
Something to note is that I do dualboot with windows 11. 

Operating System: EndeavourOS 
KDE Plasma Version: 6.2.3
KDE Frameworks Version: 6.8.0
Qt Version: 6.8.0
Kernel Version: 6.11.6-arch1-1 (64-bit)
Graphics Platform: Wayland
Processors: 16 × AMD Ryzen 7 5700G with Radeon Graphics
Memory: 15.0 GiB of RAM
Graphics Processor: AMD Radeon RX 6600
Manufacturer: Gigabyte Technology Co., Ltd.
Product Name: A520I AC
System Version: -CF

**Comment by Schievel1** [comment]
Maybe endeavor triggers the making of the grub config differently than with grub-mkconf and this different way of doing it doesn't run the 41_snapshots script.
Maybe someone running endeavor can investigate?

**Comment by KarkanAlzwayed** [comment]
Do I need to take this to the Endeavour OS team to see if it is an issue on their end?

**Comment by Schievel1** [comment]
Yes, if `sudo grub-mkconfig -o /boot/grub/grub.cfg` works, they must be doing something differently to crate the grub menu. (and therefore delete the snapshots menu while doing it)
With a quick search I couldn't find out what it is.

**Comment by dalto8** [comment]
EndeavourOS doesn't trigger the re-making of the grub menu at all.  Since the kernel/initrd names don't change on Arch/EOS with updates, there is no reason to rebuild the menu on update.

**Comment by Schievel1** [comment]
Well there must be something that triggers the making of the grub.cfg out of the usual process that includes the output of 41_snapshots-script.
Maybe some autosnapshot tool that is configured wrong. AFAIK there are a few for arch/ endeavors pacman

**Comment by dalto8** [comment]
I am not sure what the OP is using but most of our grub-btrfs users are using grub-btrfsd.

@KarkanAlzwayed, can you share what you are using?

**Comment by Schievel1** [comment]
Grub-btrfsd is triggerung grub-mkconf with exactly the same command like OP tried manually, or, if the snapshots menu entry already exists in grub.cfg, the 41_snapshots-script directly, to only trigger the generation of the snapshot submenu. 

There must be something else that triggers the creation of the grub.cfg and bypasses the usual grub-mkconf.

**Comment by KarkanAlzwayed** [comment]
> I am not sure what the OP is using but most of our grub-btrfs users are using grub-btrfsd.
> 
> @KarkanAlzwayed, can you share what you are using?

How do I get you what I'm using? Is there a command I need to run? Or would you like me to get a package name?
